docs/data/base/components/input/OTPInput.js

```diff
@@ -0,0 +1,204 @@
+          selectInput(currentIndex - 1);
+        }
+
+        setOtp((prev) => {
+          const otpArray = [...prev];
+          otpArray.splice(currentIndex, 1);
+          otpArray.push('');
+          return otpArray;
+        });
+
+        break;
+      default:
+        break;
+    }
+  };
+
+  const handleChange = (event, currentIndex) => {
+    const value = event.target.value;
+    let indexToEnter = 0;
+
+    while (indexToEnter <= currentIndex) {
+      if (otp[indexToEnter] && indexToEnter < currentIndex) {
+        indexToEnter += 1;
+      } else {
+        break;
+      }
+    }
+    setOtp((prev) => {
+      const otpArray = [...prev];
+      const lastValue = value[value.length - 1];
+      otpArray[indexToEnter] = lastValue;
+      return otpArray;
+    });
+    if (value !== '') {
+      if (currentIndex < inputCount - 1) {
+        focusInput(currentIndex + 1);
+      }
+    }
+  };
+
+  const handleClick = (event, currentIndex) => {
+    selectInput(currentIndex);
+  };
+
+  const handlePaste = (event, currentIndex) => {
+    event.preventDefault();
+    const clipboardData = event.clipboardData;
+
+    // Check if there is text data in the clipboard
+    if (clipboardData.types.includes('text/plain')) {
+      let pastedText = clipboardData.getData('text/plain');
+      pastedText = pastedText.substring(0, inputCount);
+      let indexToEnter = 0;
+
+      while (indexToEnter <= currentIndex) {
+        if (otp[indexToEnter] && indexToEnter < currentIndex) {
+          indexToEnter += 1;
+        } else {
+          break;
+        }
+      }
+
+      const otpArray = [...otp];
+
+      for (let i = indexToEnter; i < inputCount; i += 1) {
+        const lastValue = pastedText[i - indexToEnter] ?? '';
+        otpArray[i] = lastValue;
+      }
+
+      setOtp(otpArray);
+    }
+  };
+
+  return new Array(inputCount).fill(null).map((_, index) => (
+    <React.Fragment key={index}>
+      <BaseInput
+        slots={{
+          input: InputElement,
+        }}
+        aria-label={`Digit ${index + 1} of OTP`}
+        slotProps={{
+          input: {
+            ref: (ele) => {
+              inputRefs.current[index] = ele;
+            },
+            onKeyDown: (event) => handleKeyDown(event, index),
+            onChange: (event) => handleChange(event, index),
+            onClick: (event) => handleClick(event, index),
+            onPaste: (event) => handlePaste(event, index),
+            value: otp[index],
+          },
+        }}
+      />
+      {index === inputCount - 1 ? null : seperator}
+    </React.Fragment>
+  ));
+}
+
+export default function OTPInput() {
+  const inputCount = 5;
+  const [otp, setOtp] = React.useState(new Array(inputCount).fill(''));
+  return (
+    <Box sx={{ display: 'flex', gap: 1, alignItems: 'center' }}>
+      <OTP
+        seperator={<span>-</span>}
+        otp={otp}
+        setOtp={setOtp}
+        inputCount={inputCount}
+      />
+    </Box>
```

Comment:
Could we display the entered code as plain text below the OTP component?

Reply:
fixed here https://github.com/mui/material-ui/pull/40539/commits/50c9913b08e6f874e8ad6e9cddc25f5379bad2e5